### ISSUE TITLE
Not hosting Profile API causes core features to not work

### ISSUE THREAD

**Issue by DenisCarriere** [eosdac/eosdac-client-legacy#9]
Without running your own Profile API endpoint https://toolkit-api.eosdac.io

There's a lot of core features that don't work anymore:
- Get Token Balance (DAC & EOS)
- Doesn't recognize as member
- Can't register as Candidate
- Doesn't recognized signed constitution

Would it be possible to decouple the Profile API, that way some of the core features can still work by simply pulling data on chain.

The Profile API should simply add more metadata for custodians, it shouldn't prevent the member dashboard to completely stop working.

**Comment by piecesnbits** [comment]
yeah good points, however all our data is coming from that api endpoint except for the candidates (they are fetched from chain). so when disabling the api completely you won't see msig proposals, wp proposals etc.
I do think it's needed to decouple the profiles though.

**Comment by michaeljyeates** [comment]
We can probably detect a missing api and then fall back to on-chain data
